services/src/main/java/org/keycloak/organization/forms/login/freemarker/model/OrganizationAwareIdentityProviderBean.java

```diff
@@ -29,30 +28,50 @@
 
 public class OrganizationAwareIdentityProviderBean extends IdentityProviderBean {
 
-    private final IdentityProviderBean delegate;
     private final KeycloakSession session;
+    private final List<IdentityProvider> providers;
 
-    public OrganizationAwareIdentityProviderBean(IdentityProviderBean delegate, KeycloakSession session) {
-        this.delegate = delegate;
+    public OrganizationAwareIdentityProviderBean(IdentityProviderBean delegate, KeycloakSession session, boolean onlyOrganizationBrokers) {
         this.session = session;
+        if (onlyOrganizationBrokers) {
+            providers = Optional.ofNullable(delegate.getProviders()).orElse(List.of()).stream()
```

Comment:
This is nitpicky, but `Collections` offers the `emptyList()` method which also returns an immutable empty list

Reply:
I would keep `List.of`.